python/testing.rst

```diff
@@ -12,6 +12,11 @@ See :doc:`/coding/unit-test-policy` for an overview of LSST Stack testing.
 LSST tests should be written using the `unittest` framework, with default test discovery, and should support being run using the `pytest`_ test runner as well as from the command line.
 If you want to jump straight to a full example of the standard LSST Python testing boilerplate without reading the background, read the :ref:`section on memory testing <py-test-mem>` later in this document.
 
+.. note::
+   Python tests explicitly should not contain a shebang (``#!/usr/bin/env python``) and should not be executable (so cannot be run directly with ``./test_Example.py``).
+   Instead run single test files either with ``python tests/test_Example.py`` or with ``pytest tests/test_Example.py``.
```

Comment:
Why would one ever run the test file with `python` rather than `pytest`? Oh; because stack tests have an `if __name__ == "___main__" to run `unittest.main()`.

Reply:
It's *significantly* faster to run up the test if you are iterating quickly on things. Also, subTest works properly so in some cases when you are testing things it is clearer to use subTest to get all the failures from the loop in one go than it is to run pytest and get one failure then rerun and get the next failure in the loop.